lib/src/badge/Badge.tsx

```diff
@@ -1,47 +1,156 @@
-import styled, { ThemeProvider } from "styled-components";
-import useTheme from "../useTheme";
+import styled from "styled-components";
 import BadgePropsType from "./types";
+import DxcFlex from "../flex/Flex";
+import CoreTokens from "../common/coreTokens";
 
-const DxcBadge = ({ notificationText, disabled }: BadgePropsType): JSX.Element => {
-  const colorsTheme = useTheme();
+const contextualColorMap = {
+  grey: {
+    background: CoreTokens.color_grey_200,
+    text: CoreTokens.color_grey_900,
+  },
+  blue: {
+    background: CoreTokens.color_blue_200,
+    text: CoreTokens.color_blue_900,
+  },
+  green: {
+    background: CoreTokens.color_green_200,
+    text: CoreTokens.color_green_900,
+  },
+  orange: {
+    background: CoreTokens.color_orange_200,
+    text: CoreTokens.color_orange_900,
+  },
+  red: {
+    background: CoreTokens.color_red_200,
+    text: CoreTokens.color_red_900,
+  },
+  yellow: {
+    background: CoreTokens.color_yellow_200,
+    text: CoreTokens.color_yellow_900,
+  },
+  purple: {
+    background: CoreTokens.color_purple_200,
+    text: CoreTokens.color_purple_900,
+  },
+};
+
+const sizeMap = {
+  small: {
+    height: "20px",
+    width: "20px",
+    minWidth: "20px",
+    fontSize: CoreTokens.type_scale_01,
+    borderRadius: CoreTokens.spacing_12,
+    iconSize: "14px",
+    padding: {
+      contextual: `${CoreTokens.spacing_4}`,
+      notification: `${CoreTokens.spacing_0} ${CoreTokens.spacing_4}`,
+    },
+  },
+  medium: {
+    height: "24px",
+    width: "24px",
+    minWidth: "24px",
+    fontSize: CoreTokens.type_scale_02,
+    borderRadius: CoreTokens.spacing_12,
+    iconSize: "16px",
+    padding: {
+      contextual: `${CoreTokens.spacing_4} ${CoreTokens.spacing_8}`,
+      notification: `${CoreTokens.spacing_0} ${CoreTokens.spacing_4}`,
+    },
+  },
+  large: {
+    height: "32px",
+    width: "32px",
+    minWidth: "32px",
+    fontSize: CoreTokens.type_scale_04,
+    borderRadius: CoreTokens.spacing_24,
+    iconSize: "24px",
+    padding: {
+      contextual: `${CoreTokens.spacing_4} ${CoreTokens.spacing_8}`,
+      notification: `${CoreTokens.spacing_0} ${CoreTokens.spacing_8}`,
+    },
+  },
+};
+
+const Label = ({ label, notificationLimit, size }) => {
+  return (
+    <LabelContainer size={size}>
+      {typeof label === "number" ? (label > notificationLimit ? "+" + notificationLimit : label) : label}
+    </LabelContainer>
+  );
+};
+
+const DxcBadge = ({
+  label,
+  title,
+  mode = "contextual",
+  color = "grey",
+  icon,
+  notificationLimit = 99,
+  size = "medium",
+}: BadgePropsType): JSX.Element => {
   return (
-    <ThemeProvider theme={colorsTheme.tabs}>
-      <StyledDxcBadge notificationText={notificationText} disabled={disabled}>
-        <span>{notificationText}</span>
-      </StyledDxcBadge>
-    </ThemeProvider>
+    <BadgeContainer label={label} mode={mode} color={color} size={size} title={title} aria-label={title}>
+      {(mode === "contextual" && (
```

Comment:
As we check this in several places, I think we could create a flag with the verification and use it in the component. Something like:
`const showContextual = mode ==="contextual"`